test/Microsoft.Health.Fhir.Shared.Tests.E2E/Rest/DataConvertTests.cs

```diff
@@ -0,0 +1,214 @@
+﻿// -------------------------------------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License (MIT). See LICENSE in the repo root for license information.
+// -------------------------------------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using Hl7.Fhir.Model;
+using Hl7.Fhir.Rest;
+using Hl7.Fhir.Serialization;
+using Microsoft.Health.Fhir.Core.Features.Operations;
+using Microsoft.Health.Fhir.Tests.Common;
+using Microsoft.Health.Fhir.Tests.Common.FixtureParameters;
+using Microsoft.Health.Fhir.Tests.E2E.Common;
+using Microsoft.Health.Fhir.Tests.E2E.Rest;
+using Microsoft.Health.Test.Utilities;
+using Microsoft.Rest;
+using Xunit;
+using Task = System.Threading.Tasks.Task;
+
+namespace Microsoft.Health.Fhir.Shared.Tests.E2E.Rest
+{
+    /// <summary>
+    /// End to end tests using default template set only (no container registry configurations needed).
+    /// </summary>
+    [Trait(Traits.Category, Categories.DataConvert)]
+    [HttpIntegrationFixtureArgumentSets(DataStore.All, Format.Json)]
+    public class DataConvertTests : IClassFixture<HttpIntegrationTestFixture>
+    {
+        private const string DefaultTemplateSetReference = "microsofthealth/fhirconverter:default";
+        private bool _isUsingInProcTestServer = false;
+        private readonly TestFhirClient _testFhirClient;
+
+        public DataConvertTests(HttpIntegrationTestFixture fixture)
+        {
+            _isUsingInProcTestServer = fixture.IsUsingInProcTestServer;
+            _testFhirClient = fixture.TestFhirClient;
+        }
+
+        [Fact]
+        public async Task GivenAValidRequestWithDefaultTemplateSet_WhenDataConvert_CorrectResponseShouldReturn()
+        {
+            if (!_isUsingInProcTestServer)
```

Comment:
>_isUsingInProcTestServer [](start = 17, length = 24)

Why `CustomDataConvert` is getting away with not having this thing in tests, but here it's present?

Reply:
Because CustomDataConvert requires a remote FHIR server instance. Current PR check pipeline provisions a remote FHIR instance but does not enable data convert and all the tests will get "data convert not enabled" error. But since I have manually modified the templates in last commit, we can remove this check now.